src/lib.rs

```diff
@@ -596,6 +599,51 @@ fn fmt_str_opt(name: Option<&str>) -> &str {
     }
 }
 
+impl Display for CompoundTag {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
+        // Ignore self.name because it isn't accepted by Minecraft
+        // We can't use f.debug_struct() because that would use child Debug, not Display
+        write!(f, "{{")?;
+        let mut first = true;
+        for (name, value) in &self.tags {
+            write!(f, "{}{:?}:{}", if first {""} else {","}, name, value)?;
+            first = false;
+        }
+        write!(f, "}}")
+    }
+}
+
+// Display NBT in SNBT format
+impl Display for Tag {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
+        fn format_list<T: Display>(f: &mut Formatter<'_>, type_header: &'static str, list: &Vec<T>) -> Result<(), Error> {
```

Comment:
In local functions you can use arguments from main function. Will this function be inlined?

Reply:
Nested functions can't access the surrounding scope. That would require a closure, but if the closure keeps a mutable reference to the formatter, we can't use another mutable reference (for the write macros) while the closure is still life.

Reply:
Ah, indeed you are right